promql/functions.go

```diff
@@ -1270,3 +1270,34 @@ func (s *vectorByReverseValueHeap) Pop() interface{} {
 	*s = old[0 : n-1]
 	return el
 }
+
+// createLabelsForAbsentFunction returns the labels that are uniquely and exactly matched
+// in a given expression. It is used in the absent functions.
+func createLabelsForAbsentFunction(expr Expr) labels.Labels {
+	m := labels.Labels{}
+	empty := []string{}
+	lm := make([]*labels.Matcher, 0)
+
+	switch n := expr.(type) {
+	case *VectorSelector:
+		lm = n.LabelMatchers
+	case *MatrixSelector:
+		lm = n.LabelMatchers
+	}
```

Comment:
panic if it's another type?

Reply:
should we panic if it is a subquery? I dont't think so

Reply:
Let's cover that explicitly then, rather than making it seem like an oversight which could cause surprises later.

Reply:
Done but that's a lot.

Maybe we should just do:

default:
 return m

Reply:
default is fine if that's what makes sense.

Reply:
done.